org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -114,6 +118,10 @@ public void initializeProjects(final Collection<IPath> rootPaths, IProgressMonit
 		}
 		updateEncoding(monitor);
 		reportProjectsStatus();
+
+		long initializedTime = System.currentTimeMillis() - start;
+		telemetryManager.onProjectInitialized(this, initializedTime, firstTimeInitialization);
```

Comment:
In addition to track the pure project import time, we should also track the entire e2e time of service ready.

It could be `The timestamp when ServiceReady event is sent` - `The timestamp when language server starts activating`.
https://github.com/eclipse/eclipse.jdt.ls/blob/40611dc3e419821aa76f58c54314a7d23e9a059a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java#L288

Reply:
I think that makes sense. It might even possible to store all these points as timestamps given that elapsed time could be determined in post-processing.